### PR TITLE
Remove the TODO for container name matching

### PR DESCRIPTION
**Pull request by tedyu**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
In ResourceLocation, drop the TODO for when port spec starts with container name.

```release-note
NONE
```

### CODE REVIEW

**Comment by tedyu** [comment]
cc @wojtek-t

File: pkg/registry/core/pod/strategy.go [hunk 1]
```diff
@@ -247,17 +247,26 @@ func ResourceLocation(getter ResourceGetter, rt http.RoundTripper, ctx context.C
...
 	}
 
 	// Try to figure out a port.
+	parts := strings.Split(port, "/")
+	containerName := ""
+	if len(parts) == 2 {
+		containerName = parts[0]
+		port = parts[1]
+	}
+	// when port is a "(container)/(portname)", do a name lookup.
```
Comment (wojtek-t) [inline pkg/registry/core/pod/strategy.go]: Please add a unit test to test the new behavior.

Comment (tedyu) [reply pkg/registry/core/pod/strategy.go]: Searching under pkg/registry/, I still need to find some sample where ProxyREST#Connect is called.

Comment (lavalamp) [reply pkg/registry/core/pod/strategy.go]: I don't want to add more features to this proxy behavior. I would prefer to just delete the TODO. Is there a strong use case for implementing this?

**Comment by tedyu** [comment]
@lavalamp 
Can you take another look ?

**Comment by lavalamp** [comment]
/lgtm
/approve

thanks!

**Pull request merged**